sphinx/ext/autodoc/__init__.py

```diff
@@ -391,11 +391,15 @@ def format_signature(self, **kwargs: Any) -> str:
         else:
             # try to introspect the signature
             try:
-                try:
-                    args = self.format_args(**kwargs)
-                except TypeError:
-                    # retry without arguments for old documenters
-                    args = self.format_args()
+                if kwargs:
+                    try:
+                        args = self.format_args(**kwargs)
+                    except TypeError:
+                        # avoid chaining tracebacks, by putting nothing here
+                        pass
+
+                # retry without arguments for old documenters
+                args = self.format_args()
```

Comment:
It seems `format_args()` is called twice.

Reply:
I take it back, you're correct. Updated in my latest push.